services/skus/controllers.go

```diff
@@ -109,12 +109,22 @@ func Router(
 		r.Method(http.MethodPost, "/{orderID}/receipt", metricsMwr("checkOrderReceipt", authMwr(handleCheckOrderReceipt(svc, valid))))
 	}
 
+	credh := handler.NewCred(svc)
+
 	r.Route("/{orderID}/credentials", func(cr chi.Router) {
 		cr.Use(NewCORSMwr(copts, http.MethodGet, http.MethodPost))
 		cr.Method(http.MethodGet, "/", metricsMwr("GetOrderCreds", GetOrderCreds(svc)))
 		cr.Method(http.MethodPost, "/", metricsMwr("CreateOrderCreds", CreateOrderCreds(svc)))
 		cr.Method(http.MethodDelete, "/", metricsMwr("DeleteOrderCreds", authMwr(deleteOrderCreds(svc))))
 
+		// For now, this endpoint is placed directly under /credentials.
+		// It would make sense to put it under /items/item_id, had the caller known the item id.
+		// However, the caller of this endpoit does not posses that knowledge, and it would have to call the order endpoint to get it.
+		// This extra round-trip currently does not make sense.
+		// So until Bundles came along we can benefit from the fact that there is one item per order.
+		// By the time Bundles arrive, the caller would either have to fetch order anyway, or this can be communicated in another way.
+		cr.Method(http.MethodGet, "/batches/count", metricsMwr("CountBatches", authMwr(handlers.AppHandler(credh.CountBatches))))
```

Comment:
Since the AuthMwr is using the HS2019 algorithm, it seems it's possible to reuse a token from one account to authenticate the middleware here and call this against another order ID not associated with the paying customer. This seems fine because the orderID's are sufficiently random, but we need to make sure we maintain that security control in order to prevent this issue.

The alternative is to authenticate the end user, but I'm pretty sure this would defeat the privacy goals of this service if we did that.

All in all, I think this is fine as long as we keep orderID's sufficiently random. It looks like this is a UUID so would be fine. Can you point me to where the order ID value is being generated just so I can double check it?

Reply:
This is an endpoint that is used by an internal service, that's why the authentication is set up the way it is. It is similar to the other endpoints called by the same service – take a look around, for instance, at the delete credentials or set trial endpoints.

As for the way we generate UUIDs for orders, they are UUIDv4, and in this case for orders they are generated upon inserts to the database by the standard PostgresSQL extension called [`uuid-ossp`](https://www.postgresql.org/docs/14/uuid-ossp.html), specifically the function `uuid_generate_v4`.

I believe the extension's source code is available [here](http://www.ossp.org/pkg/lib/uuid/).

Also, reached out in DMs.